gr-dtv/lib/atsc/atsc_equalizer_impl.cc

```diff
@@ -148,20 +163,39 @@ int atsc_equalizer_impl::general_work(int noutput_items,
         } else {
             filterN(data_mem, data_mem2, ATSC_DATA_SEGMENT_LENGTH);
 
-            memcpy(out[output_produced].data,
+            memcpy(&out[output_produced * ATSC_DATA_SEGMENT_LENGTH],
                    data_mem2,
                    ATSC_DATA_SEGMENT_LENGTH * sizeof(float));
 
-            out[output_produced].pli._flags = d_flags;
-            out[output_produced].pli._segno = d_segno;
+            // out[output_produced].pli._flags = d_flags;
+            // out[output_produced].pli._segno = d_segno;
+            plinfo pli_out;
+            pli_out._flags = d_flags;
```

Comment:
What about endianness if using multiple hosts, or merely replaying a capture file?

Reply:
I'm not sure I follow - is this in reference to a struct being sent over std::any?

Reply:
No. The binary representation of the tag value is not portable across little/big endian. So saving a stream on a big endian machine it won't be correctly readable on a little endian machine unless they're encoded somehow. Either `htonl()` and friends, or some other serialization.

As I see it now the values are just used in the data stream as whatever the native endianness is.

Reply:
Sending a big endian packed U32 with the two values over the tags now.